pkg/rpc/server/server.go

```diff
@@ -225,7 +225,7 @@ func (s *Server) Shutdown() error {
 }
 
 func (s *Server) handleHTTPRequest(w http.ResponseWriter, httpRequest *http.Request) {
-	req := request.NewIn()
+	in := request.NewIn()
```

Comment:
Or create a batch here with one element, use that instead of `req` everywhere down below and if the request doesn't decode as a single element --- decode a batch.

Reply:
I thought about it, but how we can distinguish single-request from a batch of one request in this case?